src/foam/u2/grid/testdata/HeroModels.js

```diff
@@ -0,0 +1,213 @@
+            value: 'id',
+        },
+        {
+            name: 'groupByLabel',
+            class: 'String',
+            value: 'label',
+        },
+        /*{
+            name: 'salesforceType',
+            class: 'String'
+        },*/
+    ],
+
+});
+
+
+
+// Relationship Test
+foam.CLASS({
+  name: 'Team',
+  requires: ['Hero'], 
+  properties: [ 'name', 'id' ]
+});
+
+foam.CLASS({
+  name: 'Hero',
+  
+  requires: ['Team'],
+  
+  imports: [
+    'TeamDAO', 
+  ], 
+  
+  properties: [
+    'id',
+    {
+        name: 'fieldworker',
+        referenceProperty: 'fieldworkerId',
+        referenceDAO: 'fieldworkerDAO', 
+    },
+    
+    {
+        name: 'name',
+        gridHeaderView: function(value, obj, t){
+            return value; 
+        },
+    },
+    {
+        class: 'Date', 
+        name: 'birthDate',
+    },
+    {
+        class: 'Date', 
+        name: 'lastSeenAlive',
+        searchView: 'foam.u2.search.WeekSearchView',
+        //searchView: 'foam.u2.DateView',  
+        factory: function(){
+            return SE.util.startOfToday();
+        },
+        validator: function(obj) {
+                if (  obj.lastSeenAlive && obj.birthDate && (obj.lastSeenAlive < obj.birthDate) ) return 'Must be after birth date, unless you are a time traveller';
+        }, 
+        gridHeaderView: function(value, obj, t){
+            var d = new Date(value);
+            var p1 = foam.u2.Element.create('p');
+            p1.add(SE.util.getDayName(d));
+            var p2 = foam.u2.Element.create('p');
+            p2.add(d.getDate() + "" + SE.util.getDayCardinal(d) + " " + SE.util.getShortMonthName(d));
+            return foam.u2.Element.create('div').add(p1).add(p2); 
+        },
+    },
+    
+    {
+        name: 'status',
+        searchView: 'foam.u2.search.GroupBySearchView',
+        gridHeaderView: function(value, obj, t){
+            return value; 
+        },
+    },
+    {
+        name: 'organizationId',
+        documentation: 'organizationId should always be the one set, instead of organization object.', 
+        //hidden: true, 
+        postSet: function(old, nu){
+            console.log(this.name + " orgainizationId is set. " + nu);
+            if (!nu){
+                this.clearProperty(this.ORGANIZATION); 
+            }
+            if (nu && old!=nu){
+              this.TeamDAO.find(nu).then(function(organization) {
+                if (this.organizationId != organization.id){
+                    console.log("error. organization should be set after the organizationId, ids should match."); 
+                }
+                  this.organization = organization;
+                
+              }.bind(this));
+            }
+        }
+    },
+    
+    {
+        name: 'organization',
+        //class: 'Team', 
+        groupById: "id", 
+        groupByLabel: "name",
+        transient: true, 
+        searchView: {
+            class: 'com.serviceecho.ui.search.GroupByIdSearchView',
```

Comment:
foam code demos can't call out to 3rd part code. Either com.serviceecho.ui.search.GroupByIdSearchView would need to move to foam.* or all of this code would need to move com.serviceecho.foam.* as a 3rd party library for foam, which we could link to from the FOAM document of 3rd party FOAM libraries.